cpp/src/arrow/util/memory-pool.cc

```diff
@@ -44,14 +45,22 @@ class InternalMemoryPool : public MemoryPool {
 };
 
 Status InternalMemoryPool::Allocate(int64_t size, uint8_t** out) {
+  constexpr size_t kAlignment = 64;
   std::lock_guard<std::mutex> guard(pool_lock_);
-  *out = static_cast<uint8_t*>(std::malloc(size));
-  if (*out == nullptr) {
+  // TODO(emkornfield) find something compatible with windows
```

Comment:
Looks like `_aligned_malloc` is the thing on windows https://msdn.microsoft.com/en-us/library/8z34s9c6.aspx. since these have different APIs (the windows API sets errno on failure) we should probably factor the aligned allocation out into a utility function that returns `Status`.